posts/kubernetes-tutorial-1-intro.md

```diff
@@ -0,0 +1,69 @@
+
+To achieve this K8s changes the way we deploy applications. Instead of having dedicated hardware and manual deployment processes. Our applications have automated deployments and containerized applications. 
+
+[Kubernetes](https://kubernetes.io/) is one of many systems that enable this functionality by creating a distributed cluster of machines that all work together to host applications deployed. 
+
+## Cluster communication
+
+The way K8s operates is to have a pool of machines ready for any application to be deployed onto them. The machines communicate by having the same Kubelet application installed on all the machines. The Kubelet can manage the node it is installed on by running application containers and also has a communication endpoint for controllers nodes to send operations. 
+
+## Visualize K8s 
+
+Back to Company ABC with four application deployments. We convert our production deployment operations to k8s. First by installing a Kubelet onto all of our agent nodes. 
+
+![k8s-deployment](https://github.com/David-Elizondo/blog-usa/blob/master/images/2021/09/k8s.png)
+
+Next we segment three machines to be controllers that will be in-charge of maintaining communications with all the agent nodes and checking health of the cluster configuration. 
+
+![k8s-deployment](https://github.com/David-Elizondo/blog-usa/blob/master/images/2021/09/k8s2.png)
+
+We then define our application deployments as "Plans". The plans are defined by k8s configuration yaml files. These config files can configure many different aspects like resource sizes, network settings and health checks. These plans get communicated to our agent nodes for deployment on all active machines. 
+
+![k8s-deployment](https://github.com/David-Elizondo/blog-usa/blob/master/images/2021/09/k8s3.png)
+
+Lastly, if we repeat our disaster scenario from before and one of our machines in our agent pool is taken offline. Our controllers now can assess the health of this application automatically and redeploy our application on the next available node. 
+
+![k8s-deployment](https://github.com/David-Elizondo/blog-usa/blob/master/images/2021/09/k8s4.png)
+
+
+## Summary
+
+Using Kubernetes enables many new exciting features for Devops engineers. These functionalities are enabled by how we form our operational environment like I showed here. Using a cluster of machines we can achieve high availability. This change in deployment method allows our application deployments to withstand N machine failures where N is the size of your cluster.
+
+This high-level understanding of how clusters are formed and maintained should help your understanding of higher levels of complexities within larger and larger orchestration systems.
```

Comment:
Think about a call-to-action about how you or Ippon can help the reader with their services. Sell, sell, sell!

Reply:
Something like this? 

> Here at IPPON we have experience with designing resilient large scale application architectures. We do this by following the tech communities newest open source tools and best practices. If you are designing your next system or in charge of an Kubernetes cluster and in need of professional guidance send us over a message.

Reply:
I just deleted a long wall of text. I am going to punt this to Youssra. She will be able to come up with something better than I can.

Reply:
Pushed!